dojo/tools/stackhawk/parser.py

```diff
@@ -0,0 +1,125 @@
+        self.active = True
+        self.verified = True
+        self.static_finding = False
+        self.dynamic_finding = True
+        self.service = completed_scan['scan']['application']
+
+
+class StackHawkParser(object):
+    """
+    DAST findings from StackHawk
+    """
+
+    def get_scan_types(self):
+        return ["StackHawk HawkScan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "StackHawk HawkScan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "StackHawk webhook event can be imported in JSON format."
+
+    def get_findings(self, json_output, test):
+        completed_scan = self.__parse_json(json_output)
+
+        metadata = StackHawkScanMetadata(completed_scan)
+        findings = self.__extract_findings(completed_scan, metadata, test)
+
+        return findings
+
+    def __extract_findings(self, completed_scan, metadata: StackHawkScanMetadata, test):
+        findings = {}
+
+        if 'findings' in completed_scan:
+            raw_findings = completed_scan['findings']
+
+            for raw_finding in raw_findings:
+                key = raw_finding['pluginId']
+                if key not in findings:
+                    finding = self.__extract_finding(raw_finding, metadata, test)
+                    findings[key] = finding
+
+        # Update the test description these scan results are linked to.
+        test.description = 'View scan details here: ' + self.__hyperlink(completed_scan['scan']['scanURL'])
+
+        return list(findings.values())
+
+    def __extract_finding(self, raw_finding, metadata: StackHawkScanMetadata, test) -> Finding:
+
+        steps_to_reproduce = "Use a specific message link and click 'Validate' to see the cURL!\n\n"
+
+        host = raw_finding['host']
+        endpoints = []
+
+        for path in raw_finding['paths']:
+            steps_to_reproduce += '**' + path['path'] + '**\n' + self.__hyperlink(path['pathURL']) + '\n'
+            endpoint = Endpoint.from_uri(host + path['path'])
+            endpoints.append(endpoint)
+
+        finding = Finding(
+            test=test,
+            title=raw_finding['pluginName'],
+            date=parse_datetime(metadata.date),
+            severity=raw_finding['severity'],
+            description="View this finding in the StackHawk platform at:\n" +
+                        self.__hyperlink(raw_finding['findingURL']),
+            steps_to_reproduce=steps_to_reproduce,
+            active=metadata.active,
+            verified=metadata.verified,
+            numerical_severity=self.__convert_severity(raw_finding['severity']),
+            component_name=metadata.component_name,
+            component_version=metadata.component_version,
+            static_finding=metadata.static_finding,
+            dynamic_finding=metadata.dynamic_finding,
+            unique_id_from_tool=raw_finding['pluginId'],
+            vuln_id_from_tool=raw_finding['pluginId'],
```

Comment:
mmm. it can't be both.
if it's a "kind" of vulnerability like a rule id of the tool or a class of test it's `vuln_id_from_tool`.
if it's unique it's `unique_id_from_tool`. It can't be both.

One question that helps to find out is "can one report have the same value for raw_finding['pluginId'] ?"

Reply:
Thanks for explaining that @damiencarol! The `pluginId` maps to a rule id / class of test, and so we'll use it only as the `vuln_id_from_tool`.